lib/ContextResourceString.js

```diff
@@ -80,7 +81,8 @@ ContextResourceString.hashKey = function(project, context, locale, reskey, datat
  */
 ContextResourceString.prototype.hashKey = function() {
     var locale = this.targetLocale || this.sourceLocale;
-    return ContextResourceString.hashKey(this.project, this.context, locale, this.reskey, this.datatype, this.flavor);
+    var sourceHash = utils.hashKey(this.source);
```

Comment:
should use `this.sourceHash` instead of calculating it yourself, so that the caller can pass it in and all of the other plugins don't need to change

Reply:
Shouldn't we need sourcehash When parsing xliff file and storing data to ContextResource?
If I don't calculate sourceHash here, I can't get a correct translation data between parsed resource from qml file and translation data from xliff file. no?
* hashkey from parsed qml resource  -> `crs_i18nqmlsampler__ko-KR_energy_x-qml__r883767601`
* values from xliff -> `crs_i18nqmlsampler__ko-KR_energy_x-qml__`

Reply:
I think we are going to need a new resource string subclass: `SourceContextResourceString` This is the same as `ContextResourceString` but with the source hash in it. That way, there is no conflict between the old code and other plugins and with Qml. Then, when the xliff code reads the xliff file, it will use the ResourceFactory to make instances of the `SourceContextResourceString` which will match the type of instances that the qml plugin produces.

Reply:
We can discuss in tonight's meeting